server/lib/src/idm/identityverification.rs

```diff
@@ -0,0 +1,641 @@
+
+        assert!(idms_prox_write.qs_write.create(&ce).is_ok());
+        assert!(idms_prox_write.commit().is_ok());
+
+        let mut idms_prox_read = idms.proxy_read().await;
+
+        let ident_a = idms_prox_read
+            .qs_read
+            .internal_search_uuid(user_a_uuid)
+            .map(Identity::from_impersonate_entry_readonly)
+            .expect("Failed to impersonate identity");
+
+        let ident_b = idms_prox_read
+            .qs_read
+            .internal_search_uuid(user_b_uuid)
+            .map(Identity::from_impersonate_entry_readonly)
+            .expect("Failed to impersonate identity");
+
+        let (lower_user, lower_user_uuid, higher_user, higher_user_uuid) =
+            if user_a_uuid < user_b_uuid {
+                (ident_a, user_a_uuid, ident_b, user_b_uuid)
+            } else {
+                (ident_b, user_b_uuid, ident_a, user_a_uuid)
+            };
+
+        // First the user with the lowest uuid receives the uuid from the other user
+
+        let res_higher_user = idms_prox_read.handle_identify_user_start(
+            &IdentifyUserStartEvent::new(lower_user_uuid, higher_user.clone()),
+        );
+
+        let Ok(IdentifyUserResponse::ProvideCode { totp, .. }) = res_higher_user else {
+                return assert!(false);
+            };
+
+        let res_lower_user_wrong = idms_prox_read.handle_identify_user_submit_code(
+            &IdentifyUserSubmitCodeEvent::new(higher_user_uuid, lower_user.clone(), totp + 1),
+        );
+
+        assert!(matches!(
+            res_lower_user_wrong,
+            Ok(IdentifyUserResponse::CodeFailure)
+        ));
+
+        let res_lower_user_correct = idms_prox_read.handle_identify_user_submit_code(
+            &IdentifyUserSubmitCodeEvent::new(higher_user_uuid, lower_user.clone(), totp),
+        );
+
+        assert!(matches!(
+            res_lower_user_correct,
+            Ok(IdentifyUserResponse::ProvideCode { .. })
+        ));
+
+        // now we need to get the code from the lower_user and submit it to the higher_user
+
+        let Ok(IdentifyUserResponse::ProvideCode{totp, ..}) = res_lower_user_correct else {
+                return assert!(false);
+            };
+
+        let res_higher_user_2_wrong = idms_prox_read.handle_identify_user_submit_code(
+            &IdentifyUserSubmitCodeEvent::new(lower_user_uuid, higher_user.clone(), totp + 1),
+        );
+
+        assert!(matches!(
+            res_higher_user_2_wrong,
+            Ok(IdentifyUserResponse::CodeFailure)
+        ));
+
+        let res_higher_user_2_correct = idms_prox_read.handle_identify_user_submit_code(
+            &IdentifyUserSubmitCodeEvent::new(lower_user_uuid, higher_user.clone(), totp),
+        );
+
+        assert!(matches!(
+            res_higher_user_2_correct,
+            Ok(IdentifyUserResponse::Success)
+        ));
+    }
+
+    #[idm_test]
+
+    async fn test_codes_differ(idms: &IdmServer, _idms_delayed: &IdmServerDelayed) {
```

Comment:
Isn't this checking that? Or do you mean in some other way?

Reply:
Oh how did I miss this.